ceph_iscsi_config/client.py

```diff
@@ -114,6 +98,34 @@ def __init__(self, logger, client_iqn, image_list, username, password, mutual_us
             self.error = True
             self.error_msg = err
 
+        # image_list is normally a list of strings (pool/image_name) but
+        # group processing forces a specific lun id allocation to masked disks
+        # in this scenario the image list is a tuple
+        if image_list:
+
+            if isinstance(image_list[0], tuple):
+                # tuple format ('disk_name', {'lun_id': 0})...
+                for disk_item in image_list:
+                    disk_name = disk_item[0]
+                    lun_id = disk_item[1].get('lun_id')
+                    self.requested_images.append(disk_name)
+                    self.lun_lookup[disk_name] = lun_id
+            else:
+                for disk_name in image_list:
+                    target_config = self.config.config['targets'][self.target_iqn]
+                    disk_lun_id = target_config['disks'][disk_name]['lun_id']
+                    used_lun_ids = self._get_lun_ids(target_config['clients'])
```

Comment:
Nit: can `used_lun_ids` and `target_config` be moved outside the loop?

Reply:
Fixed.